### ISSUE TITLE
Could not obtain connection to query metadata: org.h2.jdbc.JdbcSQLNonTransientConnectionException: URL format error; must be "jdbc:h2:{ {.|mem:}[name] | [file:]fileName | {tcp|ssl}:[//]server[:port][,server2[:port]]/name }[;key=value...]" but is "jdbc:mysql

### ISSUE THREAD

**Issue by marcantoine-bibeau** [keycloak/keycloak#22805]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

operator

### Describe the bug

I updated my K8s Keycloak resource to use latest version
```
apiVersion: k8s.keycloak.org/v2alpha1
kind: Keycloak
metadata:
  name: keycloak
spec:
  instances: 1
  image: quay.io/keycloak/keycloak:latest
  db:
    vendor: mysql
    url: jdbc:mysql://ObfuscatedHost:3306/keycloak?useSSL=false&characterEncoding=UTF-8
...
```      

When starting I'm getting the following error:
```
2023-08-29 19:41:22,174 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: Base URL: <unset>, Hostname: keycloak.myhost.to, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin URL: <unset>, Admin: <request>, Port: -1, Proxied: true
2023-08-29 19:41:25,780 WARN  [io.quarkus.agroal.runtime.DataSources] (main) Datasource <default> enables XA but transaction recovery is not enabled. Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, otherwise data may be lost if the application is terminated abruptly
2023-08-29 19:41:25,969 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': URL format error; must be "jdbc:h2:{ {.|mem:}[name] | [file:]fileName | {tcp|ssl}:[//]server[:port][,server2[:port]]/name }[;key=value...]" but is "jdbc:mysql://ObfuscatedHost:3306/keycloak?useSSL=false&characterEncoding=UTF-8" [90046-220]
2023-08-29 19:41:25,978 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread) HHH000342: Could not obtain connection to query metadata: org.h2.jdbc.JdbcSQLNonTransientConnectionException: URL format error; must be "jdbc:h2:{ {.|mem:}[name] | [file:]fileName | {tcp|ssl}:[//]server[:port][,server2[:port]]/name }[;key=value...]" but is "jdbc:mysql://ObfuscatedHost:3306/keycloak?useSSL=false&characterEncoding=UTF-8" [90046-220]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:690)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.engine.ConnectionInfo.getFormatException(ConnectionInfo.java:687)
	at org.h2.engine.ConnectionInfo.<init>(ConnectionInfo.java:86)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:113)
	at org.h2.jdbcx.JdbcDataSource.getXAConnection(JdbcDataSource.java:322)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Version

from 22.0.1 to latest (and nightly)

### Expected behavior

pod should start without exception...

### Actual behavior

Pod not starting with mentioned exception

### How to Reproduce?

Having keycloak properly working on version 22.0.1, update the Keycloak spec with `image: quay.io/keycloak/keycloak:latest`

**Comment by shawkins** [comment]
@marcantoine-bibeau the issue here is the explicit use of the default image.  If image is specified the logic assumes an optimized one is in use and passes --optimized.  If you omit the image from the cr, it should work as you expect.

Is there a reason you need to explicitly specify the default, and/or are you thinking that there should be a check of if the image matches the default?

**Comment by marcantoine-bibeau** [comment]
Hi @shawkins, The reason I wanted to specify the image is to use the latest version that have a bug fix for [this](https://github.com/keycloak/keycloak/pull/21627) that I wanted to use before there's a now release

**Comment by marcantoine-bibeau** [comment]
How does the Operator knows about optimized image or not? 
Would it be possible to publish an "optimized" image of latest?

**Comment by shawkins** [comment]
> How does the Operator knows about optimized image or not?

Whatever is configured as the default operator image is supposed to be unoptimized.  If an image is referenced in a Keycloak CR it is assumed to be optimized

> Would it be possible to publish an "optimized" image of latest?

Are you asking if the keycloak project would do this or how to do it as a user?  It is not likely something that the keycloak project will publish - an optimized / custom image is configured to use you database of choice and other setting so that start up will be faster.

**Comment by marcantoine-bibeau** [comment]
But what is the difference of me using the Operator version 22.0.1 that create operator Statefulset/pods using `image: quay.io/keycloak/keycloak:22.0.1-1`? I assume that `quay.io/keycloak/keycloak:22.0.1-1` is not optimized? Because when pod starts I can see that all features a present 
```
Installed features: [agroal, cdi, hibernate-orm, jdbc-h2, jdbc-mariadb, jdbc-mssql, jdbc-mysql, jdbc-oracle, jdbc-postgresql, keycloak, logging-gelf, micrometer, narayana-jta, reactive-routes, resteasy, resteasy-jackson, smallrye-context-propagation, smallrye-health, vertx
```
What I'm starting to understand is that I should build my own Keycloak image to only have the feature I want and not everything, good :)

If I cannot override the "image" in the Keycloak operator CRD ressource, I would have expected to be able to use the "main" branch version (kinda latest) but if I use `https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/main/kubernetes/kubernetes.yml`, it install `quay.io/keycloak/keycloak-operator:18.0.2`

So, in short, I'm using the operator, how can I run main/latest/nightly images without having to build my own image? (That I will eventually when my PoC is conclusive ;))

**Comment by shawkins** [comment]
> So, in short, I'm using the operator, how can I run main/latest/nightly images without having to build my own image? (That I will eventually when my PoC is conclusive ;))

See this comment - https://github.com/keycloak/keycloak/issues/22805#issuecomment-1701142397 - you need to tell the operator what you want as the default image.

**Comment by marcantoine-bibeau** [comment]
Ha! `https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/main/kubernetes/kubernetes.yml` refer to 18.0.2 as I mentioned in my last comment BUT `https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/nightly/kubernetes/kubernetes.yml` seems ok and uses 999.0.0-SNAPSHOT so I can try the bug fix I was waiting for!

**Comment by vmuzikar** [comment]
It may make sense for `keycloak-k8s-resources/main` to point to nightly.
